sanic/mixins/startup.py

```diff
@@ -880,12 +880,17 @@ def serve(
             manager.run()
         except ServerKilled:
             exit_code = 1
+        except SanicException as e:
+            kwargs = {}
+            if e.quiet:
+                error_logger.error(str(e))
+            else:
+                raise
         except BaseException:
             kwargs = primary_server_info.settings
             error_logger.exception(
                 "Experienced exception while trying to serve"
             )
-            raise
```

Comment:
Why? With this change, won't it exit with 0?

Reply:
That is what raising would achieve. But I get your point assuming you have console logging on. exit here is fine as long as it is `exit(1)`.

Reply:
This is what `raise` causes (on sanic main):

```
[2023-09-07 14:41:52 +0100] [481] [ERROR] Experienced exception while trying to serve
Traceback (most recent call last):
  File "/home/user/sanic/sanic/server/socket.py", line 110, in configure_socket
    sock = bind_socket(
  File "/home/user/sanic/sanic/server/socket.py", line 33, in bind_socket
    sock.bind(location)
PermissionError: [Errno 13] Permission denied

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/user/sanic/sanic/mixins/startup.py", line 1033, in serve
    for sock in [
  File "/home/user/sanic/sanic/mixins/startup.py", line 1034, in <listcomp>
    configure_socket(server_info.settings)
  File "/home/user/sanic/sanic/server/socket.py", line 125, in configure_socket
    raise error
sanic.exceptions.ServerError: Sanic server could not start: [Errno 13] Permission denied.

This may have happened if you are running Sanic in the global scope and not inside of a `if __name__ == "__main__"` block.

See more information: https://sanic.dev/en/guide/deployment/manager.html#how-sanic-server-starts-processes

[2023-09-07 14:41:52 +0100] [481] [INFO] Server Stopped
Traceback (most recent call last):
  File "/home/user/sanic/sanic/server/socket.py", line 110, in configure_socket
    sock = bind_socket(
  File "/home/user/sanic/sanic/server/socket.py", line 33, in bind_socket
    sock.bind(location)
PermissionError: [Errno 13] Permission denied

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/sanic", line 33, in <module>
    sys.exit(load_entry_point('sanic', 'console_scripts', 'sanic')())
  File "/home/user/sanic/sanic/__main__.py", line 12, in main
    cli.run(args)
  File "/home/user/sanic/sanic/cli/app.py", line 117, in run
    serve(app)
  File "/home/user/sanic/sanic/mixins/startup.py", line 1033, in serve
    for sock in [
  File "/home/user/sanic/sanic/mixins/startup.py", line 1034, in <listcomp>
    configure_socket(server_info.settings)
  File "/home/user/sanic/sanic/server/socket.py", line 125, in configure_socket
    raise error
sanic.exceptions.ServerError: Sanic server could not start: [Errno 13] Permission denied.

This may have happened if you are running Sanic in the global scope and not inside of a `if __name__ == "__main__"` block.

See more information: https://sanic.dev/en/guide/deployment/manager.html#how-sanic-server-starts-processes
```

Notice how the full traceback is printed twice? First by `error_logger.exception`, then there is `raise` and then CPython catches Sanic crashing and prints another traceback (without the logger).

Using `sys.exit(1)` instead of `raise` avoids this.